crates/storage/provider/Cargo.toml

```diff
@@ -43,3 +43,8 @@ tempfile = "3.3"
 
 [features]
 test-utils = ["reth-rlp"]
+optimism = [
+  "reth-primitives/optimism",
+  "reth-interfaces/optimism",
+  "reth-revm-primitives/optimism"
```

Comment:
All these feature deps indicate to me we're missing optimism-specific code in this crate